vm/vm.go

```diff
@@ -60,23 +80,28 @@ func (vm *VM) SetupGenesis(genesis *config.GenesisConfig) error {
 // ApplyLayer applies the given rewards to some miners as well as a vector of
 // transactions for the given layer. to miners vector for layer. It returns an
 // error on failure, as well as a vector of failed transactions.
-func (vm *VM) ApplyLayer(lid types.LayerID, txs []*types.Transaction, rewards []types.AnyReward) ([]*types.Transaction, error) {
+func (vm *VM) ApplyLayer(lid types.LayerID, txs []*types.Transaction, rewards []types.AnyReward) ([]*types.Transaction, []*types.Reward, error) {
```

Comment:
i think types.Reward should be removed from Apply signature

Reply:
i don't understand this comment. do you mean that we only keep rewards data inside vm?